app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -704,12 +707,32 @@ public SQLiteDatabase getHandle() {
         return currentApp.getStorage(name, c);
     }
 
+    public <T extends Persistable> HybridFileBackedSqlStorage<T> getFileBackedAppStorage(String name, Class<T> c) {
+        return currentApp.getFileBackedStorage(name, c);
+    }
+
     public <T extends Persistable> SqlStorage<T> getUserStorage(Class<T> c) {
         return getUserStorage(c.getAnnotation(Table.class).value(), c);
     }
 
     public <T extends Persistable> SqlStorage<T> getUserStorage(String storage, Class<T> c) {
-        return new SqlStorage<>(storage, c, new AndroidDbHelper(this.getApplicationContext()) {
+        return new SqlStorage<>(storage, c, buildUserDbHandle());
+    }
+
+    public <T extends Persistable> HybridFileBackedSqlStorage<T> getFileBackedUserStorage(String storage, Class<T> c) {
+        return new HybridFileBackedSqlStorage<>(storage, c, buildUserDbHandle(), getUserKeyRecordId());
+    }
+
+    public String getUserKeyRecordId() {
+        try {
+            return getSession().getUserKeyRecordUUID();
+        } catch (SessionUnavailableException e) {
+            throw new RuntimeException(e);

```

Comment:
Using the user key record uuid as the subdirectory name for a specific user level file-backed db directory. Made sure this same uuid is also used when performing db migrations from the old db setup to this new one.